Reading List/Reading List/Views/ReadingListTableViewController.swift

```diff
@@ -0,0 +1,163 @@
+    }
+ 
+
+    /*
+    // Override to support conditional editing of the table view.
+    override func tableView(_ tableView: UITableView, canEditRowAt indexPath: IndexPath) -> Bool {
+        // Return false if you do not want the specified item to be editable.
+        return true
+    }
+    */
+
+  
+    // Override to support editing the table view.
+    override func tableView(_ tableView: UITableView, commit editingStyle: UITableViewCell.EditingStyle, forRowAt indexPath: IndexPath) {
+        if editingStyle == .delete {
+            let deletedBook = bookFor(indexPath: indexPath)
+            bookController.Delete(bookToDelete: deletedBook)
+            tableView.deleteRows(at: [indexPath], with: .fade)
+        }
+    }
+    
+
+    /*
+    // Override to support rearranging the table view.
+    override func tableView(_ tableView: UITableView, moveRowAt fromIndexPath: IndexPath, to: IndexPath) {
+
+    }
+    */
+
+    /*
+    // Override to support conditional rearranging of the table view.
+    override func tableView(_ tableView: UITableView, canMoveRowAt indexPath: IndexPath) -> Bool {
+        // Return false if you do not want the item to be re-orderable.
+        return true
+    }
+    */
+
+    /*
+    // MARK: - Navigation
+
+    // In a storyboard-based application, you will often want to do a little preparation before navigation
+    override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
+        // Get the new view controller using segue.destination.
+        // Pass the selected object to the new view controller.
+    }
+    */
+    override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
+        // Get the new view controller using segue.destination.
+        
+        switch segue.identifier{
+        case "addDetailSegue":
+            let detailVC = segue.destination as! BookDetailViewController
+            detailVC.bookController = bookController
+        case "cellDetailSegue":
+            let cell = sender as! UITableViewCell
+            let indexPath = tableView.indexPath(for: cell)
+            let detailVC = segue.destination as! BookDetailViewController
+            let book = bookFor(indexPath: indexPath!)
+            detailVC.bookController = bookController
+            detailVC.book = book
+        case .none:
+            return
+        case .some(_):
+            return
+        }
+        return
+    }
+    
+    
+    //Conditional Section Checking Logic
+    
+    func bookFor(indexPath: IndexPath) -> Book {
+        if indexPath.section == 0 {
+            return bookController.readBooks[indexPath.row]
+        } else {
+            return bookController.unReadBooks[indexPath.row]
+        }
+        
+    }
+    
+    
+    // Delegate Func Implementation
+    
+    
+ //   TODO: [Need to refactor this to find the correct indexPath for needed book]
+    
+    func toggleHasBeenRead(for cell: BookTableViewCell) {
+        guard let book = cell.book else { return }
```

Comment:
Whenever you need a `book` object you should access it through `bookController`
```suggestion
        guard let indexPath = tableView.indexPath(for: cell) else { return }
        let book = bookFor(indexPath: indexPath)
```